RFC-0009-native-tpcds-connector.md

```diff
@@ -0,0 +1,46 @@
+# **RFC-0009 for Presto**
+
+## Presto - Native TPC-DS connector
+
+Proposers
+
+* Pratik Joseph Dabre
+* Pramod Satya
+
+## [Related Issues]
+
+Related issues: https://github.com/prestodb/presto/issues/22361
+
+Related PRs: https://github.com/prestodb/presto/pull/23067
+
+## Summary
+
+A native TPC-DS connector capable of generating in-memory data on the fly is proposed.
+
+## Background
+
+Currently , Presto does not have a native implementation of the TPC-DS connector. This RFC proposes the addition of a new TPC-DS connector. The new connector can be used as a Presto - Native catalog.
+
+### [Optional] Goals
+
+1. Add a TPC-DS connector to generate TPC-DS data in Presto native. 
+2. Write end-to-end tests in Presto native with TPC-DS tables and conduct microbenchmarks in Velox. 
+
+### [Optional] Non-goals
+
+## Proposed Implementation
+
+The Presto - Native TPC-DS connector will be a wrapper for the generator distributed (dsdgen) by the TPC organization from C. This means we need our implementation to have the exact same behavior as the C implementation. DuckDB already has a TPC-DS connector of their own and they have wrapped the C files into C++ files, we are going to use these C++ files in our implementation.
+
+In the C++ implementation, there are two types of tables: source tables and target tables used for generation. Source table files are prefixed with "s_", while target table files are prefixed with "w_". For instance, there may be files like "s_call_center.c" and "w_call_center.c". It appears that source tables are only utilized when running the "dsdgen" with an update flag, though the exact function of this flag and the purpose of the source tables have not yet been explored. Currently, our focus is solely on implementing functionalities for the target tables (w_ tables).
```

Comment:
Doesn't Ying want to generate the update data with TPC-DS connector as well ? Might be good to add more information about it here.